tfhe/c_api_tests/test_high_level_compact_list.c

```diff
@@ -0,0 +1,129 @@
+#include "tfhe.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+int main(void) {
+  int ok = 0;
+
+  // First, we create a ClientKey and a CompactPublicKey
+  ClientKey *client_key = NULL;
+  CompactPublicKey *public_key = NULL;
+  {
+    ConfigBuilder *builder;
+    Config *config;
+    ok = config_builder_default(&builder);
+    assert(ok == 0);
+    ok = config_builder_build(builder, &config);
+    assert(ok == 0);
+
+    ok = client_key_generate(config, &client_key);
+    assert(ok == 0);
+
+    ok = compact_public_key_new(client_key, &public_key);
+    assert(ok == 0);
+  }
+
+  // Then, we create the compact list
+  CompactCiphertextList *compact_list = NULL;
+  {
+    CompactCiphertextListBuilder *builder;
+    ok = compact_ciphertext_list_builder_new(public_key, &builder);
+    assert(ok == 0);
+
+    // Push some values
+    ok = compact_ciphertext_list_builder_push_u32(builder, 38382);
+    assert(ok == 0);
+
+    ok = compact_ciphertext_list_builder_push_i64(builder, -1);
+    assert(ok == 0);
+
+    ok = compact_ciphertext_list_builder_push_bool(builder, true);
+    assert(ok == 0);
+
+    ok = compact_ciphertext_list_builder_push_u2(builder, 3);
+    assert(ok == 0);
+
+    ok = compact_ciphertext_list_builder_build(builder, &compact_list);
+    assert(ok == 0);
+
+    // Don't forget to destroy the builder
+    compact_ciphertext_list_builder_destroy(builder);
+  }
+
+  // Now we can expand values
+  FheUint32 *a = NULL;
+  FheInt64 *b = NULL;
+  FheBool *c = NULL;
+  FheUint2 *d = NULL;
+  {
+    FheTypes type = Fhe_Bool;
```

Comment:
might be a tad error prone to have FheBool be the actual type and Fhe_Bool an enum value I would think ?

Reply:
Should we have Type_Fhe_Bool ?

Reply:
What's the usual convention for C API enum names ? this one I'm guessing ?

Then yeah I guess Type_Fhe_Bool or Type_FheBool